packages/Search/src/DTK_FineSearchFunctor.hpp

```diff
@@ -0,0 +1,361 @@
+    {
+        return Intrepid2::Impl::Triangle<3>::checkPointInclusion(
+            reference_point, threshold );
+    }
+};
+
+class Triangle_6
+{
+  public:
+    template <typename DeviceType>
+    KOKKOS_INLINE_FUNCTION void
+    mapToReferenceFrame( Kokkos::View<Coordinate *, DeviceType> reference_point,
+                         Kokkos::View<Coordinate *, DeviceType> physical_point,
+                         Kokkos::View<Coordinate **, DeviceType> nodes ) const
+    {
+        typedef Intrepid2::Impl::Basis_HGRAD_TRI_C2_FEM basis_type;
+        Intrepid2::Impl::CellTools::Serial::mapToReferenceFrame<basis_type>(
+            reference_point, physical_point, nodes );
+    }
+
+    template <typename DeviceType>
+    KOKKOS_INLINE_FUNCTION bool
+    checkPointInclusion( Kokkos::View<Coordinate *, DeviceType> reference_point,
+                         double threshold ) const
+    {
+        return Intrepid2::Impl::Triangle<6>::checkPointInclusion(
+            reference_point, threshold );
+    }
+};
+
+class Wedge_6
+{
+  public:
+    template <typename DeviceType>
+    KOKKOS_INLINE_FUNCTION void
+    mapToReferenceFrame( Kokkos::View<Coordinate *, DeviceType> reference_point,
+                         Kokkos::View<Coordinate *, DeviceType> physical_point,
+                         Kokkos::View<Coordinate **, DeviceType> nodes ) const
+    {
+        typedef Intrepid2::Impl::Basis_HGRAD_WEDGE_C1_FEM basis_type;
+        Intrepid2::Impl::CellTools::Serial::mapToReferenceFrame<basis_type>(
+            reference_point, physical_point, nodes );
+    }
+
+    template <typename DeviceType>
+    KOKKOS_INLINE_FUNCTION bool
+    checkPointInclusion( Kokkos::View<Coordinate *, DeviceType> reference_point,
+                         double threshold ) const
+    {
+        return Intrepid2::Impl::Wedge<6>::checkPointInclusion( reference_point,
+                                                               threshold );
+    }
+};
+
+class Wedge_18
+{
+  public:
+    template <typename DeviceType>
+    KOKKOS_INLINE_FUNCTION void
+    mapToReferenceFrame( Kokkos::View<Coordinate *, DeviceType> reference_point,
+                         Kokkos::View<Coordinate *, DeviceType> physical_point,
+                         Kokkos::View<Coordinate **, DeviceType> nodes ) const
+    {
+        typedef Intrepid2::Impl::Basis_HGRAD_WEDGE_C2_FEM basis_type;
+        Intrepid2::Impl::CellTools::Serial::mapToReferenceFrame<basis_type>(
+            reference_point, physical_point, nodes );
+    }
+
+    template <typename DeviceType>
+    KOKKOS_INLINE_FUNCTION bool
+    checkPointInclusion( Kokkos::View<Coordinate *, DeviceType> reference_point,
+                         double threshold ) const
+    {
+        return Intrepid2::Impl::Wedge<18>::checkPointInclusion( reference_point,
+                                                                threshold );
+    }
+};
+}
+
+namespace Functor
+{
+template <typename CellType, typename DeviceType>
```

Comment:
I am missing something. In `FineSearch::search` you decide which of the Hexahedron8/Hexahedron27/... to use based on the key. For each of them you have a unique basis function as is hardcoded in `Hexahedron_8` and friends. So why is it not possible to move all this into the search functor? Would something like this work?
```c++
template <typename DeviceType>
class FineSearch
{
  public:
    FineSearch(
       ...
        shards::CellTopology cell_topo )
    {
      // assign all views
      switch ( cell_topo.getKey() )
      {
      case hex_8.key:
         _cell = Intrepid2::Impl::Hexahedron<8>;
        _basis = Intrepid2::Impl::Basis_HGRAD_HEX_C1_FEM();
    }

    KOKKOS_INLINE_FUNCTION
    void operator()( unsigned int const i ) const
    {
      ...
     Intrepid2::Impl::CellTools::Serial::mapToReferenceFrame<decltype(_basis)>(
            reference_point, physical_point, nodes );
        _point_in_cell[i] =
            decltype(_cell)::.checkPointInclusion( ref_point, _threshold );
    }
};
```